internal/domain/power-status.go

```diff
@@ -1078,3 +1103,40 @@ func startPowerStatusMaster(lastUpdated time.Time) {
 		}
 	}
 }
+
+func updateSmdPowerState(updatedComponents []pcsmodel.PowerStatusComponent) error {
+	// The states we care about propagating to SMD, currently only On and Off?
+	// Looking at the SMD code, it implies that only On and Off should be updated
+	// via endpoints, the likes of Unknown, Empty are "internal" states.
+	stateToPropagate := []string{pcsmodel.PowerStateFilter_On.String(), pcsmodel.PowerStateFilter_Off.String()}
+
+	// Split the updated components by state
+	compsByState := make(map[string][]string)
+	for _, state := range stateToPropagate {
+		compsByState[state] = make([]string, 0)
+	}
+
+	for _, comp := range updatedComponents {
+		if _, ok := compsByState[comp.PowerState]; ok {
+			compsByState[comp.PowerState] = append(compsByState[comp.PowerState], comp.XName)
+		}
+	}
+
+	// For each state, do a bulk update to SMD
+	// Note if the number of components becomes excessive, we can do a series of
+	// batch updates here.
+	for state, components := range compsByState {
+		// Skip if no components to update
+		if len(components) == 0 {
+			continue
+		}
+
+		// Update SMD for components that are on
```

Comment:
This isn't actually filtering updates by state, and we didn't intend to only send updates for `On` components, correct? Since we're not doing any additional filtering past the above map, probably fine to omit any additional comment here.

```suggestion
```

Reply:
Left over from before the loop 😄